File: src/types/ai.ts
// Re-export generated types from ts-rs
export type { AiProvider, GenerateSqlRequest } from "./generated";

// Frontend-only type (not in Rust, stored in localStorage)
export interface AiSettings {
  provider: "Claude" | "Ollama";
  claudeApiKey: string;
  ollamaBaseUrl: string;
  ollamaModel: string;
}

File: src-tauri/src/types/ai.rs
//! AI-related types

use serde::{Deserialize, Serialize};
use ts_rs::TS;

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[ts(export, export_to = "../../src/types/generated/")]
pub enum AiProvider {
    Claude,
    Ollama,
}

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[ts(export, export_to = "../../src/types/generated/")]
pub struct GenerateSqlRequest {
    pub prompt: String,
    pub schema_context: String,
    pub provider: AiProvider,
    #[ts(optional)]
    pub api_key: Option<String>,
    #[ts(optional)]
    pub ollama_base_url: Option<String>,
    #[ts(optional)]
    pub ollama_model: Option<String>,
}
